contracts/utility/FantomDeFiTokenStorage.sol

```diff
@@ -109,52 +123,88 @@ contract FantomDeFiTokenStorage is Initializable, IFantomDeFiTokenStorage
     }
 
     // totalOfDec returns the value of current balance of an account
     // with specified token balance decreased by given amount of tokens.
-    function totalOfDec(address _account, address _token, uint256 _amount) external view returns (uint256 value) {
+    function totalOfDec(
+        address _account,
+        address _token,
+        uint256 _amount,
+        bool requireTradable
+    ) external view returns (uint256 value) {
         // calculate the total with token balance adjusted down
-        return _totalOf(_account, _token, 0, _amount);
+        return _totalOf(_account, _token, 0, _amount, requireTradable);
     }
 
     // balanceOf returns the balance of the given token on the given account.
-    function balanceOf(address _account, address _token) public view returns (uint256) {
+    function balanceOf(address _account, address _token)
+        public
+        view
+        returns (uint256)
+    {
         return balance[_account][_token];
     }
 
     // _totalOf calculates the value of given account with specified token balance adjusted
     // either up, or down, based on given extra values
-    function _totalOf(address _account, address _token, uint256 _add, uint256 _sub) internal view returns (uint256 value) {
+    function _totalOf(
+        address _account,
+        address _token,
+        uint256 _add,
+        uint256 _sub,
+        bool requireTradable
+    ) internal view returns (uint256 value) {
         // loop all registered debt tokens
-        for (uint i = 0; i < tokens.length; i++) {
+        for (uint256 i = 0; i < tokens.length; i++) {
             // advance the result by the value of current token balance of this token.
             // Make sure to stay on safe size with the _sub deduction, we don't
             // want to drop balance to sub-zero amount, that would freak out the SafeMath.
             if (_token == tokens[i]) {
+                uint256 adjustedBalance = balance[_account][tokens[i]]
+                    .add(_add)
+                    .sub(_sub, "token sub exceeds balance");
                 // add adjusted token balance converted to value
                 // NOTE: this may revert on underflow if the _sub value exceeds balance,
                 // but it should never happen on normal protocol operations.
-                value = value.add(tokenValue(
-                        tokens[i],
-                        balance[_account][tokens[i]].add(_add).sub(_sub, "token sub exceeds balance")
-                    ));
+                value = value.add(tokenValue(tokens[i], adjustedBalance));
 
                 // we consumed the adjustment and can reset it
                 _add = 0;
                 _sub = 0;
             } else {
-                // simply add the token balance converted to value as-is
-                value = value.add(tokenValue(tokens[i], balance[_account][tokens[i]]));
+                if (requireTradable) {
```

Comment:
if (!requireTradable || addressProvider.getTokenRegistry().canTrade(tokens[i]))

Reply:
I've tried this way. Doesn't let the test pass!